### ISSUE TITLE
No interactive plotting with "Widget" in JupyterLab 3.8.3

### ISSUE THREAD

**Issue by joeheaven** [winpython/winpython#880]
Hi,

I just installed a fresh WinPython 3.8.3. It seems that everything is fine.

I'm trying to achieve some interaction with my plots in Jupyter, which is a brilliant feature and I use it a lot.

1. Try Jupyter Notebook:

I open a regular Jupyter (not Lab) instance and run this code:

```python
import matplotlib.pyplot as plt
import numpy as np
%matplotlib notebook
x = np.linspace(0, 10, 100)
plt.plot(x, np.sin(x))
plt.plot(x, np.cos(x))
plt.show()
```

and I get a very nice plot that I can play with.

2. Try JupyterLab:

I open a JupyterLab instance and run the following code:

```python
import matplotlib.pyplot as plt
import numpy as np
%matplotlib widget
x = np.linspace(0, 10, 100)
plt.plot(x, np.sin(x))
plt.plot(x, np.cos(x))
plt.show()
```

I receive the follwowing message:

```
Warning: Cannot change to a different GUI toolkit: widget. Using notebook instead.
Javascript Error: IPython is not defined
```

Please note that I changed from backend notebook to widget.

I know that this issue is also present in Anaconda, because there the necessary extensions were not installed, but if I check with WinPython JupyterLab I can see that the `@jupyter-widgets/jupyterlab-manager` is installed.

I guess/hope that I'm just missing something.

In the release notes for 3.8.3 there is a note: `Jupyterlab-2.1.2, Qt-5.14.2, python-vscode-2020.5.80290 (bug... it now requires a .py file to have been opened and ran before a .ipynb is open with ipywidgets working)` which I honestly don't understand.

Any help or hint would be highly appreciated, because I would very much move back from Anaconda to WinPython.

**Comment by stonebig** [comment]
What shall works is what is tested in winpthon_checket.ipynb

**Comment by stonebig** [comment]
Put %matplotlib widget on 1st line ? And restart kernel before trying

**Comment by stonebig** [comment]
And rinse twice the cell if no success at 1st run

**Comment by stonebig** [comment]
Remove last blank line

**Comment by stonebig** [comment]
If your pc is slow/tiny, not win10, it may not work well

**Comment by joeheaven** [comment]
Dear @stonebig, all your hints were so most valuable. 

tl;dr: All fine now! Thanks a lot!

Longer feedback: In fact I first started by trying the example notebook. Except numba (which was be expected not to run on my laptop), everything works absolutely fine.

I then tried to start again my tiny piece of code, and after implementing everything you said above: move magic to first line, include another magic later on (hint from example notebook), restarting the kernel I got the graph expected, with all controls and all worked.

Thank you very much, this was really bothering me since we changed from WinPython to Anaconda half a year back, because of mixed Win/Lin environment, but such a lot didn't work out of the box as we were used to in WinPython I'm evaluating to switch back to WinPython. The interactivity in lab was a major requirement, which is now fulfilled.

Fantastic help.

**Comment by stonebig** [comment]
great, so we can close this issue
